app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -75,8 +81,14 @@ class CiYaml {
     List<Target> enabledTargets = _filterEnabledTargets(postsubmitTargets);
     // Filter targets removed from main.
     if (totPostsubmitTargetNames!.isNotEmpty) {
-      enabledTargets =
-          enabledTargets.where((Target target) => totPostsubmitTargetNames!.contains(target.value.name)).toList();
+      final String defaultBranch = Config.defaultBranch(slug);
+      enabledTargets = enabledTargets
+          .where(
+            (Target target) =>
+                !target.value.enabledBranches.contains(defaultBranch) ||
```

Comment:
Do you need to check `target.value.enabledBranches.isNotEmpty` as you do for `presubmitTargets`? Nit: create a sub function to share the logic.

Reply:
Added the validation to postsubmit targets. Only a portion of the condition can be shared, <1 line.

Reply:
All these lines are sharable. A sub function is more clear to me, but not a big issue and up to you.

```
enabledTargets = getEnabledTargets(slug, enabledTargets, totPresubmitTargetNames);
enabledTargets = getEnabledTargets(slug, enabledTargets, totPostsubmitTargetNames);

getEnabledTargets(slug, enabledTargets, totTargetNames){
  final String defaultBranch = Config.defaultBranch(slug);
  return enabledTargets
          .where(
            (Target target) =>
                (target.value.enabledBranches.isNotEmpty && !target.value.enabledBranches.contains(defaultBranch)) ||
                totTargetNames!.contains(target.value.name),
          )
          .toList();
}

Reply:
Done